distribution/docker/build.gradle

```diff
@@ -402,11 +411,11 @@ void addBuildEssDockerImageTask(Architecture architecture) {
     tasks.register(taskName("build", architecture, base, "DockerImage"), DockerBuildTask) {
 
       TaskProvider<DockerBuildTask> buildCloudTask = tasks.named(taskName("build", architecture, DockerBase.CLOUD, "DockerImage"))
-      dependsOn(buildCloudTask)
-      dependsOn(buildContextTask)
+      inputs.file(buildCloudTask.map({ it.markerFile }))
```

Comment:
This can be simplified to just be `inputs.files(buildCloudTask)`. Gradle will automatically add all task output as an input to this task.